### PR TITLE
refactor(LB/certificate): update lb_certificate with v2 api

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
**What this PR does / why we need it**:

update lb_certificate with v2 api

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #NONE

**Special notes for your reviewer**:

**Release note**:

```release-note
1. update the API version and remove openstack API
```

## PR Checklist

* [x] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud' TESTARGS='-run TestAccLBV2Certificate'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud -v -run TestAccLBV2Certificate -timeout 360m -parallel 4
=== RUN   TestAccLBV2Certificate_basic
=== PAUSE TestAccLBV2Certificate_basic
=== RUN   TestAccLBV2Certificate_client
=== PAUSE TestAccLBV2Certificate_client
=== CONT  TestAccLBV2Certificate_basic
=== CONT  TestAccLBV2Certificate_client
--- PASS: TestAccLBV2Certificate_client (69.86s)
--- PASS: TestAccLBV2Certificate_basic (113.40s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       113.486s
```

### CODE REVIEW

File: huaweicloud/resource_huaweicloud_lb_certificate.go [hunk 1]
```diff
@@ -85,66 +95,71 @@ func ResourceCertificateV2() *schema.Resource {
...
-	d.Set("type", c.Type)
-	d.Set("domain", c.Domain)
-	d.Set("certificate", c.Certificate)
-	d.Set("private_key", c.PrivateKey)
-	d.Set("create_time", c.CreateTime)
-	d.Set("update_time", c.UpdateTime)
-
-	d.Set("region", GetRegion(d, config))
+	mErr := multierror.Append(nil,
+		d.Set("region", config.GetRegion(d)),
```
Comment (chengxiangdong) [inline huaweicloud/resource_huaweicloud_lb_certificate.go:138]: does `enterprise_project_id` need to be setted value

File: huaweicloud/resource_huaweicloud_lb_certificate.go [hunk 2]
```diff
@@ -85,66 +95,71 @@ func ResourceCertificateV2() *schema.Resource {
...
+	elbClient, err := config.LoadBalancerClient(GetRegion(d, config))
 	if err != nil {
-		return fmtp.Errorf("Error creating HuaweiCloud elb client: %s", err)
+		return fmtp.DiagErrorf("Error creating HuaweiCloud elb client: %s", err)
 	}
 
 	c, err := certificates.Get(elbClient, d.Id()).Extract()
 	if err != nil {
-		return CheckDeleted(d, err, "certificate")
+		return common.CheckDeletedDiag(d, err, "certificate")
```
Comment (chengxiangdong) [inline huaweicloud/resource_huaweicloud_lb_certificate.go]: the exception information `certificate` is not clear, the log printed is: certificate: xxx

File: huaweicloud/resource_huaweicloud_lb_certificate_test.go [hunk 3]
```diff
@@ -292,3 +315,74 @@ EOT
...
+
+func testAccLBV2CertificateConfig_withEpsId(name string) string {
+	return fmt.Sprintf(`
+resource "huaweicloud_lb_certificate" "certificate_1" {
+  name                  = "%s"
+  description           = "terraform test certificate"
+  domain                = "www.elb.com"
+  enterprise_project_id = "%s"
+  private_key           = <<EOT
+-----BEGIN RSA PRIVATE KEY-----
```
Comment (chengxiangdong) [inline huaweicloud/resource_huaweicloud_lb_certificate_test.go]: Where did you get this certificate? Will there be a risk of leakage?

File: huaweicloud/resource_huaweicloud_lb_certificate.go [hunk 4]
```diff
@@ -85,66 +95,71 @@ func ResourceCertificateV2() *schema.Resource {
...
+		Name:                d.Get("name").(string),
+		Description:         d.Get("description").(string),
+		Type:                d.Get("type").(string),
+		Domain:              d.Get("domain").(string),
+		PrivateKey:          d.Get("private_key").(string),
+		Certificate:         d.Get("certificate").(string),
+		EnterpriseProjectID: GetEnterpriseProjectID(d, config),
 	}
 
 	logp.Printf("[DEBUG] Create Options: %#v", createOpts)
```
Comment (chengxiangdong) [inline huaweicloud/resource_huaweicloud_lb_certificate.go:107]: privacy and security issue, do not print the sensitive arguments `private_key` and `certificate` to log

Comment (chengxiangdong) [reply huaweicloud/resource_huaweicloud_lb_certificate.go:107]: Also in the update method

File: huaweicloud/resource_huaweicloud_lb_certificate.go [hunk 5]
```diff
@@ -72,6 +75,13 @@ func ResourceCertificateV2() *schema.Resource {
 				DiffSuppressFunc: utils.SuppressNewLineDiffs,
```
Comment (chengxiangdong) [inline huaweicloud/resource_huaweicloud_lb_certificate.go:76]: `certificate` and `private_key` are sensitive arguments, suggest add the attribute with `Sensitive` as true.

File: huaweicloud/resource_huaweicloud_lb_certificate.go [hunk 6]
```diff
@@ -72,6 +75,13 @@ func ResourceCertificateV2() *schema.Resource {
...
 			},
 
+			"enterprise_project_id": {
```
Comment (chengxiangdong) [inline huaweicloud/resource_huaweicloud_lb_certificate.go]: pls add a description in the doc

File: huaweicloud/resource_huaweicloud_lb_certificate.go (hunk 1 above)
Comment (Jason-Zhang9309) [reply huaweicloud/resource_huaweicloud_lb_certificate.go:138]: enterprise_project_id is not contained in api response

File: huaweicloud/resource_huaweicloud_lb_certificate.go (hunk 4 above)
Comment (Jason-Zhang9309) [reply huaweicloud/resource_huaweicloud_lb_certificate.go:107]: fixed

File: huaweicloud/resource_huaweicloud_lb_certificate.go (hunk 5 above)
Comment (Jason-Zhang9309) [reply huaweicloud/resource_huaweicloud_lb_certificate.go:76]: fixed

File: huaweicloud/resource_huaweicloud_lb_certificate.go (hunk 6 above)
Comment (Jason-Zhang9309) [reply huaweicloud/resource_huaweicloud_lb_certificate.go]: sure

File: huaweicloud/resource_huaweicloud_lb_certificate_test.go (hunk 3 above)
Comment (Jason-Zhang9309) [reply huaweicloud/resource_huaweicloud_lb_certificate_test.go]: This certificate has been in the test case for a long time. It seems the certificate is just for testing and there is no risk.

**Pull request merged**